include/solvers/unsteady_solver.h

```diff
@@ -150,6 +150,38 @@ class UnsteadySolver : public TimeSolver
    */
   virtual bool is_steady() const libmesh_override { return false; }
 
+  /**
+   * Indicate to the UnsteadySolver which variables are first order in time.
+   * Subclasses may use this information in different ways.
+   */
+  void add_first_order_var(unsigned int var)
+  { _first_order_vars.insert(var); }
+
```

Comment:
This whole commit feels redundant; if we extend its return value to be unsigned int instead of bool, can't we query TimeSolver::system()::is_time_evolving()?  https://en.wikipedia.org/wiki/Don't_repeat_yourself isn't a hard rule but there should usually be a good reason for breaking it and I'm not sure what that is here.